SecretManager/src/V1/Client/BaseClient/SecretManagerServiceBaseClient.php

```diff
@@ -0,0 +1,674 @@
+use Google\Cloud\SecretManager\V1\GetSecretRequest;
+use Google\Cloud\SecretManager\V1\GetSecretVersionRequest;
+use Google\Cloud\SecretManager\V1\ListSecretVersionsRequest;
+use Google\Cloud\SecretManager\V1\ListSecretsRequest;
+use Google\Cloud\SecretManager\V1\Secret;
+use Google\Cloud\SecretManager\V1\SecretVersion;
+use Google\Cloud\SecretManager\V1\UpdateSecretRequest;
+use GuzzleHttp\Promise\PromiseInterface;
+
+/**
+ * Service Description: Secret Manager Service
+ *
+ * Manages secrets and operations using those secrets. Implements a REST
+ * model with the following objects:
+ *
+ * * [Secret][google.cloud.secretmanager.v1.Secret]
+ * * [SecretVersion][google.cloud.secretmanager.v1.SecretVersion]
+ *
+ * This class provides the ability to make remote calls to the backing service through method
+ * calls that map to API methods.
+ *
+ * Many parameters require resource names to be formatted in a particular way. To
+ * assist with these names, this class includes a format method for each type of
+ * name, and additionally a parseName method to extract the individual identifiers
+ * contained within formatted names that are returned by the API.
+ *
+ * This class is currently experimental and may be subject to changes.
+ *
+ * @internal
+ *
+ * @method PromiseInterface accessSecretVersionAsync(AccessSecretVersionRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface addSecretVersionAsync(AddSecretVersionRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface createSecretAsync(CreateSecretRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface deleteSecretAsync(DeleteSecretRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface destroySecretVersionAsync(DestroySecretVersionRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface disableSecretVersionAsync(DisableSecretVersionRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface enableSecretVersionAsync(EnableSecretVersionRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface getIamPolicyAsync(GetIamPolicyRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface getSecretAsync(GetSecretRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface getSecretVersionAsync(GetSecretVersionRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface listSecretVersionsAsync(ListSecretVersionsRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface listSecretsAsync(ListSecretsRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface setIamPolicyAsync(SetIamPolicyRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface testIamPermissionsAsync(TestIamPermissionsRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface updateSecretAsync(UpdateSecretRequest $request, array $optionalArgs = [])
+ */
+class SecretManagerServiceBaseClient
```

Comment:
Just a reminder we still want to mark this `abstract`

Reply:
I will update this once https://github.com/googleapis/gapic-generator-php/pull/601 goes in.

Reply:
Done!